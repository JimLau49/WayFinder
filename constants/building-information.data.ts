export const BuildingInformation = [
  //SGW campus
  {
    building: "Hall",
    name: "Henry F. Hall Building",
    address: "1455 De Maisonneuve Blvd O., Montreal, QC, H3G 1M8"
  },
  {
    building: "LB",
    name: "JW McConnell Building",
    address: "1400 De Maisonneuve Blvd. O. Montreal, QC, H3G 1M8"
  },

  {
    building: "GN",
    name: "Grey Nuns Building",
    address: "1190 Guy St, Montreal, Quebec H3H 2L4"
  },

  {
    building: "GS",
    name: "GS Building",
    address: "1538 Sherbrooke St W, Montreal, Quebec H3G 1L5"
  },
  {
    building: "EV",
    name: "EV Building",
    address: "1493-1515 Saint-Catherine St W, Montreal, Quebec H3G 2W1"
  },

  {
    building: "LD",
    name: "LD Building",
    address:
      "2050 Mackay, 2070 Mackay St, Montreal, QC H3G 2J1 (Has a bunch of student buildings)"
  },
  {
    building: "TD",
    name: "TD Building",
    address:
      "2050 Mackay, 2070 Mackay St, Montreal, QC H3G 2J1 (Concordia University Student Parents Centre (CUSP))"
  },
  {
    building: "FG",
    name: "FG Building",
    address: "1616 Saint-Catherine St W, Montreal, Quebec H3H 1L7"
  },
  {
    building: "JMSB",
    name: "John Molson School of Buisness",
    address: "1450 Guy St, Montreal, Quebec H3H 0A1"
  },

  {
    building: "VA",
    name: "Visual Arts Building",
    address: "1395 René-Lévesque Blvd W, Montreal, QC H3G 2M5"
  },

  {
    building: "SAM",
    name: "Samuel Bronfman Building",
    address: "1590 Docteur Penfield, Montreal, QC"
  },
  //LOYOLA Campus
  {
    building: "RJComplex",
    name: "Richard J. Renaud Science Complex",
    address:
      "Richard J. Renaud Science Complex, 3475 Rue West Broadway, Montreal, QC H4B 2A7"
  },
  {
    building: "Terrebonne",
    name: "Terrebonne Building",
    address: "7079 Terrebonne, 7075 Rue de Terrebonne, Montreal, QC H4B 1E1"
  },
  {
    building: "VanierLibrary",
    name: "Vanier Library Building",
    address: "7141 Sherbrooke St W, Montreal, Quebec H4B 1R6"
  },
  {
    building: "CJ",
    name: "Communications & Journalism building",
    address: "Notre-Dame-de-Grâce, Montreal, QC H4B 1R6"
  },
  {
    building: "CC",
    name: "Central building",
    address: "Notre-Dame-de-Grâce, Montreal, QC H4B 1R6"
  },
  {
    building: "AD",
    name: "Administration building",
    address: "Notre-Dame-de-Grâce, Montreal, QC H4B 1R6"
  },
  {
    building: "PS",
    name: "Physical Serices building",
    address: "Notre-Dame-de-Grâce, Montreal, QC H4B 2B9"
  },
  {
    building: "RA",
    name: "Recreation and Athletic Complex",
    address: "Gym, 2 Path to gym, Montreal, QC H4B 1R2"
  },
  {
    building: "PC",
    name: "PERFORM Centre",
    address: "7200 Sherbrooke St W Montreal, QC H4B 1R2"
  },
  {
    building: "GE",
    name: "Centre for Structural and Functional Genomics",
    address: "7141 Rue Sherbrooke Ouest, Montréal, H4B 1R6"
  }
];
