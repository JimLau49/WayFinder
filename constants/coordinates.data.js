export const CampusCoordinates = {
  SGW: {
    latitude: 45.495869,
    longitude: -73.578107,
    latitudeDelta: 0.0222,
    longitudeDelta: 0.0121
  },
  loyola: {
    latitude: 45.4584,
    longitude: -73.64045,
    latitudeDelta: 0.0222,
    longitudeDelta: 0.0121
  }
};

export const BuildingCoordinates = {
  // SGW campus
  Hall: [
    { latitude: 45.49771, longitude: -73.579034 },
    { latitude: 45.497373, longitude: -73.578339 },
    { latitude: 45.496832, longitude: -73.57885 },
    { latitude: 45.497164, longitude: -73.579543 }
  ],
  LB: [
    { latitude: 45.496708, longitude: -73.578525 },
    { latitude: 45.497291, longitude: -73.578017 },
    { latitude: 45.496896, longitude: -73.577102 },
    { latitude: 45.496255, longitude: -73.5777 },
    { latitude: 45.496708, longitude: -73.578525 }
  ],
  GN: [
    { latitude: 45.492616, longitude: -73.576544 },
    { latitude: 45.493853, longitude: -73.575514 },
    { latitude: 45.494425, longitude: -73.577054 },
    { latitude: 45.493429, longitude: -73.578126 }
  ],
  EV: [
    { latitude: 45.49511, longitude: -73.577861 },
    { latitude: 45.495465, longitude: -73.577501 },
    { latitude: 45.49599, longitude: -73.57849 },
    { latitude: 45.495596, longitude: -73.5788 }
  ],
  JMSB: [
    { latitude: 45.495195, longitude: -73.578535 },
    { latitude: 45.494988, longitude: -73.57876 },
    { latitude: 45.495323, longitude: -73.579401 },
    { latitude: 45.495523, longitude: -73.579196 }
  ],
  FG: [
    { latitude: 45.493626, longitude: -73.578718 },
    { latitude: 45.493817, longitude: -73.579072 },
    { latitude: 45.494686, longitude: -73.578034 },
    { latitude: 45.494457, longitude: -73.577636 }
  ],
  GS: [
    { latitude: 45.496421, longitude: -73.580935 },
    { latitude: 45.4966, longitude: -73.581461 },
    { latitude: 45.496784, longitude: -73.581303 },
    { latitude: 45.496566, longitude: -73.580804 }
  ],
  LD: [
    { latitude: 45.497087, longitude: -73.579927 },
    { latitude: 45.496675, longitude: -73.578983 },
    { latitude: 45.496545, longitude: -73.579115 },
    { latitude: 45.49698, longitude: -73.580014 }
  ],
  TD: [
    { latitude: 45.495057, longitude: -73.57828 },
    { latitude: 45.494978, longitude: -73.578139 },
    { latitude: 45.494943, longitude: -73.578176 },
    { latitude: 45.495023, longitude: -73.578332 }
  ],
  VA: [
    { latitude: 45.496192, longitude: -73.573781 },
    { latitude: 45.495667, longitude: -73.574304 },
    { latitude: 45.49541, longitude: -73.573748 },
    { latitude: 45.495665, longitude: -73.573482 },
    { latitude: 45.495813, longitude: -73.573807 },
    { latitude: 45.496079, longitude: -73.573544 }
  ],
  SAM: [
    { latitude: 45.496446, longitude: -73.586218 },
    { latitude: 45.496683, longitude: -73.586074 },
    { latitude: 45.496506, longitude: -73.58569 },
    { latitude: 45.49646, longitude: -73.585711 }
  ],

  //LOYOLA
  RJComplex: [
    { latitude: 45.458468, longitude: -73.64132 },
    { latitude: 45.458318, longitude: -73.640863 },
    { latitude: 45.457528, longitude: -73.641468 },
    { latitude: 45.457201, longitude: -73.640653 },
    { latitude: 45.456963, longitude: -73.640853 },
    { latitude: 45.457448, longitude: -73.642094 }
  ],
  Terrebonne: [
    { latitude: 45.460082, longitude: -73.640892 },
    { latitude: 45.459977, longitude: -73.640967 },
    { latitude: 45.45994, longitude: -73.640865 },
    { latitude: 45.460041, longitude: -73.640774 }
  ],
  VanierLibrary: [
    { latitude: 45.459109, longitude: -73.637828 },
    { latitude: 45.458608, longitude: -73.638351 },
    { latitude: 45.459036, longitude: -73.639505 },
    { latitude: 45.459525, longitude: -73.639165 }
  ],
  CJ: [
    { latitude: 45.457831, longitude: -73.640486 },
    { latitude: 45.457477, longitude: -73.639675 },
    { latitude: 45.457069, longitude: -73.640057 },
    { latitude: 45.45737, longitude: -73.640879 },
    { latitude: 45.457858, longitude: -73.640548 }
  ],
  CC: [
    { latitude: 45.458218, longitude: -73.639895 },
    { latitude: 45.45808, longitude: -73.640005 },
    { latitude: 45.458381, longitude: -73.640789 },
    { latitude: 45.458521, longitude: -73.640679 }
  ],
  AD: [
    { latitude: 45.457902, longitude: -73.640122 },
    { latitude: 45.458382, longitude: -73.639744 },
    { latitude: 45.458262, longitude: -73.63945 },
    { latitude: 45.457792, longitude: -73.639825 }
  ],
  PS: [
    { latitude: 45.459634, longitude: -73.639192 },
    { latitude: 45.459285, longitude: -73.639457 },
    { latitude: 45.459709, longitude: -73.640322 },
    { latitude: 45.459985, longitude: -73.640136 }
  ],
  RA: [
    { latitude: 45.456394, longitude: -73.637372 },
    { latitude: 45.456727, longitude: -73.637101 },
    { latitude: 45.456727, longitude: -73.637101 },
    { latitude: 45.457065, longitude: -73.637993 },
    { latitude: 45.456749, longitude: -73.638252 }
  ],
  PC: [
    { latitude: 45.45669, longitude: -73.636975 },
    { latitude: 45.456954, longitude: -73.636774 },
    { latitude: 45.457289, longitude: -73.637637 },
    { latitude: 45.457022, longitude: -73.637836 }
  ],
  GE: [
    { latitude: 45.456947, longitude: -73.640738 },
    { latitude: 45.457184, longitude: -73.640564 },
    { latitude: 45.457044, longitude: -73.640165 },
    { latitude: 45.456801, longitude: -73.640348 }
  ]
};
