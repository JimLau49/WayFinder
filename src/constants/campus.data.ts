import { Campus, CampusId, MarkerLocation } from "../types/main";

/**
 * Converts a given campus to a marker location.
 *
 * MarkerLocation is used for the start and end locations for a travel plan,
 * and can be a building, POI or campus.
 *
 * @param campus Campus object to convert to a marker location
 */
export const campusToMarker = (campus: Campus): MarkerLocation => {
  return {
    id: `campus-${campus.id}`,
    displayName: campus.id,
    location: campus.region,
  };
};

/**
 * Get Campus by id.
 * @param id Id of the campus
 */
export const getCampusById = (id: CampusId): Campus => {
  return Campuses.filter((campus) => campus.id === id)[0];
};

/**
 * Get All campuses
 */
export const getAllCampuses = (): Campus[] => {
  return Campuses;
};

const Campuses: Campus[] = [
  {
    id: "SGW",
    displayName: "Sir George Williams Campus",
    description: "",
    region: {
      latitude: 45.495376377001215,
      longitude: -73.5777054168284,
      latitudeDelta: 0.013275900094058102,
      longitudeDelta: 0.009238533675670624,
    },
    buildings: [
      "B",
      "CB",
      "CI",
      "CL",
      "D",
      "EN",
      "ER",
      "EV",
      "FA",
      "FB",
      "FG",
      "GA",
      "GM",
      "GN",
      "GS",
      "H",
      "K",
      "LB",
      "LD",
      "LS",
      "M",
      "MB",
      "MI",
      "MU",
      "P",
      "PR",
      "Q",
      "R",
      "RR",
      "S",
      "SB",
      "T",
      "TD",
      "V",
      "VA",
      "X",
      "Z",
    ],
  },
  {
    id: "Loyola",
    displayName: "Loyola Campus",
    description: "",
    region: {
      latitude: 45.45840118394234,
      latitudeDelta: 0.013102120988982335,
      longitude: -73.63942572847009,
      longitudeDelta: 0.009821243584156036,
    },
    buildings: [
      "AD",
      "BB",
      "BH",
      "CC",
      "CJ",
      "DO",
      "FC",
      "GE",
      "HA",
      "HB",
      "HC",
      "HU",
      "JR",
      "PC",
      "PS",
      "PT",
      "PY",
      "QA",
      "RA",
      "RF",
      "SC",
      "SH",
      "SI",
      "SP",
      "TA",
      "VE",
      "VL",
    ],
  },
];
