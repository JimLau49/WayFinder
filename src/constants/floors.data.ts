import { BuildingFloor } from "../types/main";

export const buildingFloors: BuildingFloor[] = [
  {
    id: 1,
    buildingId: "H",
    level: 8,
    bounds: null,
    image: null,
    travelNodes: [
      {
        id: 0,
        location: {
          latitude: 45.497411422038525,
          longitude: -73.57930387743083,
        },
        children: [1],
      },
      {
        id: 1,
        location: {
          latitude: 45.49735509062119,
          longitude: -73.57918419446206,
        },
        children: [0, 2, 15, 17],
      },
      {
        id: 2,
        location: {
          latitude: 45.49743264359791,
          longitude: -73.57911244537091,
        },
        children: [1, 3, 4],
      },
      {
        id: 3,
        location: {
          latitude: 45.497389166942305,
          longitude: -73.57901186253285,
        },
        children: [2],
      },
      {
        id: 4,
        location: {
          latitude: 45.49756354338001,
          longitude: -73.57899677510714,
        },
        children: [2, 5, 9],
      },
      {
        id: 5,
        location: {
          latitude: 45.497613365120166,
          longitude: -73.57895318921065,
        },
        children: [4],
      },
      {
        id: 6,
        location: {
          latitude: 45.497348275354476,
          longitude: -73.57855018730616,
        },
        children: [7, 9],
      },
      {
        id: 7,
        location: {
          latitude: 45.49714193688946,
          longitude: -73.57874800022101,
        },
        children: [6, 8, 10, 11],
      },
      {
        id: 8,
        location: {
          latitude: 45.49708224429956,
          longitude: -73.57862964774823,
        },
        children: [7],
      },
      {
        id: 9,
        location: {
          latitude: 45.49741149279668,
          longitude: -73.57868966217494,
        },
        children: [4, 6, 13],
      },
      {
        id: 10,
        location: {
          latitude: 45.49720844470052,
          longitude: -73.57888278122401,
        },
        children: [7, 13, 15],
      },
      {
        id: 11,
        location: {
          latitude: 45.496983774888996,
          longitude: -73.57889082785105,
        },
        children: [7, 12],
      },
      {
        id: 12,
        location: { latitude: 45.49719904430793, longitude: -73.579332051234 },
        children: [11, 17],
      },
      {
        id: 13,
        location: {
          latitude: 45.49736449098771,
          longitude: -73.57873593028044,
        },
        children: [9, 10, 14],
      },
      {
        id: 14,
        location: {
          latitude: 45.497409612725065,
          longitude: -73.57882980759597,
        },
        children: [13],
      },
      {
        id: 15,
        location: {
          latitude: 45.497301508502154,
          longitude: -73.57906181867575,
        },
        children: [1, 10, 16],
      },
      {
        id: 16,
        location: { latitude: 45.49733534984642, longitude: -73.5790262794063 },
        children: [15],
      },
      {
        id: 17,
        location: {
          latitude: 45.49730597420072,
          longitude: -73.57923445639382,
        },
        children: [1, 12, 18],
      },
      {
        id: 18,
        location: {
          latitude: 45.49726038236117,
          longitude: -73.57913990852605,
        },
        children: [17],
      },
    ],
    disabledUnfriendly: [],
  },
  {
    id: 2,
    buildingId: "H",
    level: 9,
    bounds: null,
    image: null,
    travelNodes: [
      {
        id: 0,
        location: {
          latitude: 45.497348,
          longitude: -73.578553,
        },
        children: [1, 12],
      },
      {
        id: 1,
        location: {
          latitude: 45.497154,
          longitude: -73.578732,
        },
        children: [2, 11],
      },
      {
        id: 2,
        location: {
          latitude: 45.496979,
          longitude: -73.578896,
        },
        children: [1, 3],
      },
      {
        id: 3,
        location: {
          latitude: 45.497081,
          longitude: -73.579109,
        },
        children: [2, 4, 5],
      },
      {
        id: 4,
        location: {
          latitude: 45.497042,
          longitude: -73.579145,
        },
        children: [3],
      },
      {
        id: 5,
        location: {
          latitude: 45.497146,
          longitude: -73.579202,
        },
        children: [3, 6, 9],
      },
      {
        id: 6,
        location: {
          latitude: 45.497219,
          longitude: -73.579361,
        },
        children: [5, 7],
      },
      {
        id: 7,
        location: {
          latitude: 45.497283,
          longitude: -73.579301,
        },
        children: [6, 8],
      },
      {
        id: 8,
        location: {
          latitude: 45.497277,
          longitude: -73.579226,
        },
        children: [7, 9],
      },
      {
        id: 9,
        location: {
          latitude: 45.497233,
          longitude: -73.579128,
        },
        children: [5, 8, 10],
      },
      {
        id: 10,
        location: {
          latitude: 45.497322,
          longitude: -73.579066,
        },
        children: [9, 11, 13],
      },
      {
        id: 11,
        location: {
          latitude: 45.497223,
          longitude: -73.578861,
        },
        children: [1, 10, 12],
      },
      {
        id: 12,
        location: {
          latitude: 45.497414,
          longitude: -73.578683,
        },
        children: [0, 11],
      },
      {
        id: 13,
        location: {
          latitude: 45.49739,
          longitude: -73.579014,
        },
        children: [10, 14, 17],
      },
      {
        id: 14,
        location: {
          latitude: 45.497454,
          longitude: -73.579139,
        },
        children: [13, 15],
      },
      {
        id: 15,
        location: {
          latitude: 45.497489,
          longitude: -73.57917,
        },
        children: [14, 16],
      },
      {
        id: 16,
        location: {
          latitude: 45.497651,
          longitude: -73.579018,
        },
        children: [15, 17],
      },
      {
        id: 17,
        location: {
          latitude: 45.497563,
          longitude: -73.578837,
        },
        children: [13, 16, 18],
      },
      {
        id: 18,
        location: {
          latitude: 45.497504,
          longitude: -73.578713,
        },
        children: [17],
      },
    ],
    disabledUnfriendly: [],
  },
  {
    id: 3,
    buildingId: "MB",
    level: 1,
    bounds: [
      [45.495778, -73.5797],
      [45.494802, -73.578297],
    ],
    image: require("../../assets/floors/MB-1.png"),
    travelNodes: [
      {
        id: 0,
        location: {
          latitude: 45.49527159782319,
          longitude: -73.57945036143064,
        },
        children: [1],
      },
      {
        id: 1,
        location: {
          latitude: 45.49538722650027,
          longitude: -73.57932228595018,
        },
        children: [0, 11],
      },
      {
        id: 2,
        location: {
          latitude: 45.495498624635324,
          longitude: -73.57925221323967,
        },
        children: [3, 11],
      },
      {
        id: 3,
        location: {
          latitude: 45.49536466482581,
          longitude: -73.57903562486172,
        },
        children: [2, 4, 6],
      },
      {
        id: 4,
        location: {
          latitude: 45.495250211192285,
          longitude: -73.57914458960295,
        },
        children: [3, 5],
      },
      {
        id: 5,
        location: {
          latitude: 45.49527089276958,
          longitude: -73.57918918132782,
        },
        children: [4],
      },
      {
        id: 6,
        location: {
          latitude: 45.49536913015795,
          longitude: -73.57889883220196,
        },
        children: [3, 12],
      },
      {
        id: 7,
        location: {
          latitude: 45.49522388917384,
          longitude: -73.57863295823336,
        },
        children: [8, 9, 12, 14],
      },
      {
        id: 8,
        location: {
          latitude: 45.49526219710733,
          longitude: -73.57857998460531,
        },
        children: [7],
      },
      {
        id: 9,
        location: {
          latitude: 45.495190281578864,
          longitude: -73.57858702540398,
        },
        children: [7],
      },
      {
        id: 10,
        location: {
          latitude: 45.49516513462401,
          longitude: -73.57874594628811,
        },
        children: [7],
      },
      {
        id: 11,
        location: {
          latitude: 45.495454035943894,
          longitude: -73.57927702759403,
        },
        children: [1, 2],
      },
      {
        id: 12,
        location: {
          latitude: 45.495319,
          longitude: -73.578834,
        },
        children: [6, 7, 13],
      },
      {
        id: 13,
        location: {
          latitude: 45.495291,
          longitude: -73.578866,
        },
        children: [12, 14],
      },
      {
        id: 14,
        location: {
          latitude: 45.495197,
          longitude: -73.578708,
        },
        children: [7, 10, 13],
      },
    ],
    disabledUnfriendly: [7, 12],
  },
  {
    id: 4,
    buildingId: "CC",
    level: 1,
    bounds: [
      [45.458639, -73.640921],
      [45.457962, -73.639775],
    ],
    image: require("../../assets/floors/CC1.png"),
    travelNodes: [
      {
        id: 0,
        location: {
          latitude: 45.45844095574591,
          longitude: -73.64070363342762,
        },
        children: [1],
      },
      {
        id: 1,
        location: {
          latitude: 45.45836076219479,
          longitude: -73.64047598093748,
        },
        children: [0, 2],
      },
      {
        id: 2,
        location: {
          latitude: 45.45825399406411,
          longitude: -73.64020708948374,
        },
        children: [1, 3],
      },
      {
        id: 3,
        location: {
          latitude: 45.458161806440636,
          longitude: -73.63997407257557,
        },
        children: [2, 4],
      },
      {
        id: 4,
        location: {
          latitude: 45.45813687812879,
          longitude: -73.6399157345295,
        },
        children: [3],
      },
    ],
    disabledUnfriendly: [],
  },
];
