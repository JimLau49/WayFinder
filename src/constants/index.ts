export * from "./buildings.data";
export * from "./campus.data";
export * from "./floors.data";
export * from "./poi.data";
