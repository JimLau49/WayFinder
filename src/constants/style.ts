// Height of the Campus Toggle Button
export const CAMPUS_TOGGLE_HEIGHT = 80;

// Colors
export const CONCORDIA_RED = "#AA2B45";
export const BUILDING_UNTAPPED = "#dba4a8";
export const INACTIVE_BUTTON_COLOR = "#F2F2F2";
