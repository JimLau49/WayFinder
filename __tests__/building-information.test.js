import React from "react";
import BuildingInformation from "../src/components/building-information/building-information.component";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { getBuildingById } from "../src/constants";

describe("BuildingInformation component", () => {
  it("should match snapshot", () => {
    const tree = renderer.create(<BuildingInformation />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should call onClosePanel on 'x' button press", () => {
    const mockOnClosePanel = jest.fn();
    const wrapper = shallow(
      <BuildingInformation
        tappedBuilding={"H"}
        showBuildingInfo={true}
        onClosePanel={mockOnClosePanel}
      />
    );
    const button = wrapper.find("TouchableOpacity");
    // NOTE: Cannot currently test
    // Waiting for answer: https://github.com/octopitus/rn-sliding-up-panel/issues/165
    // button.simulate("press");
  });

  it("should call startBuildingTravelPlan on start travel button press", () => {
    const mockStartBuildingTravelPlan = jest.fn();
    const wrapper = shallow(
      <BuildingInformation
        tappedBuilding={"H"}
        showBuildingInfo={true}
        onClosePanel={null}
        startBuildingTravelPlan={mockStartBuildingTravelPlan}
      />
    );
    wrapper
      .find({ testID: "building-info-start-travel" })
      .first()
      .props()
      .onPressOut();

    expect(mockStartBuildingTravelPlan).toHaveBeenCalledTimes(1);
    expect(mockStartBuildingTravelPlan).toHaveBeenCalledWith(
      getBuildingById("H")
    );
  });
});
