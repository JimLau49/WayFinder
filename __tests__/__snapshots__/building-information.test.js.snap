// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildingInformation component should match snapshot 1`] = `
<View
  disableScrollViewPanResponder={true}
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "alignSelf": "stretch",
      "bottom": -1292,
      "flex": 1,
      "height": 1292,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "transform": Array [
        Object {
          "translateY": -113,
        },
      ],
    }
  }
/>
`;
